### PR TITLE
[move source language] Added support for signer

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Added signer as a primitive resource type
- Added move_to builtin
- Added rules around transaction arguments
- Blocked by #3913, so only look at the top commit 

## Test Plan

- migrated tests

### CODE REVIEW

File: language/move-lang/src/hlir/translate.rs [hunk 1]
```diff
@@ -1183,33 +1186,73 @@ fn use_tmp(var: Var) -> H::UnannotatedExp_ {
     }
 }
 
+// let (bf, arg) = match *bf {
```
Comment (sblackshear) [inline language/move-lang/src/hlir/translate.rs:1189]: Intentional?

File: language/move-lang/src/typing/translate.rs [hunk 2]
```diff
@@ -90,32 +90,50 @@ fn script(context: &mut Context, nscript: N::Script) -> T::Script {
...
-        format_comma(&["u8", "u64", "u128", "bool", "address", "vector<u8>"]),
-    );
-    context.error(vec![(mloc, mmsg), (*loc, tmsg)])
+    if idx != 0 && subtype_no_report(context, ty.clone(), signer_ref.clone()).is_ok() {
+        let mmsg = format!(
+            "Invalid parameter for script function '{}'",
+            context.current_function.as_ref().unwrap()
+        );
+        let tmsg = format!(
+            "{} is only supported for the first argument to a script",
```
Comment (sblackshear) [inline language/move-lang/src/typing/translate.rs:131]:
```suggestion
            "{} must be the first argument to a script",
```

File: language/move-lang/src/hlir/translate.rs (hunk 1 above)
Comment (tnowacki) [reply language/move-lang/src/hlir/translate.rs]: Good catch! Removed

**Pull request closed without merging**